xdsl/printer.py

```diff
@@ -362,10 +365,11 @@ def print_attribute(self, attribute: Attribute) -> None:
             return
 
         if isinstance(attribute, DictionaryAttr):
+            data = cast(DictionaryAttr[Attribute], attribute).data
             self.print_string("{")
-            self.print_dictionary(
-                attribute.data,  # type: ignore
-                self.print_attribute)
+            # The keys are attributes, even if type is annotated as dict[str, Attribute]
```

Comment:
I don't understand this comment. Isn't it fixed by this PR?

Reply:
Nope, that was part of my question, I'm happy to fix it in this PR also